ct-app/core/subgraph/graphql_provider.py

```
@@ -0,0 +1,226 @@
+    def _initialize_query(
+        self, query_file: str, extra_inputs: Optional[list[str]] = None
+    ):
+        if extra_inputs is None:
+            extra_inputs = []
+
+        keys, self._sku_query = self._load_query(query_file, extra_inputs)
+
+        if self.default_key is None:
+            self.default_key = keys
+
+    def _load_query(self, path: Union[str, Path], extra_inputs: list[str] = []) -> str:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)